src/shared/codelens/codeLensUtils.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import { LambdaHandlerCandidate } from '../lambdaHandlerSearch'
+import {
+    SamCliProcessInvoker,
+    SamCliTaskInvoker
+} from '../sam/cli/samCliInvoker'
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+export const toolkitOutputChannel: vscode.OutputChannel = vscode.window.createOutputChannel(
+    localize('AWS.channel.aws.toolkit', 'AWS Toolkit')
+)
+export const log = (msg: string): void => {
+    toolkitOutputChannel.show()
+    toolkitOutputChannel.appendLine(msg)
+}
+
+export interface CodeLensProviderParams {
+    configuration: SettingsConfiguration,
+    toolkitOutputChannel: vscode.OutputChannel,
+    processInvoker?: SamCliProcessInvoker,
+    taskInvoker?: SamCliTaskInvoker
+}
+
+export const getLambdaHandlerCandidates = async ({uri}: {uri: vscode.Uri}): Promise<LambdaHandlerCandidate[]> => {
+    const filename = uri.fsPath
+    // DocumentSymbol[]> => {
+    if (!vscode.window.activeTextEditor) {
+        log('Oh no, vscode.window.activeTextEditor is not defined!')
+
+        return []
+    } else {
+        log(`getting symbols for ${uri.fsPath}`)
+        const symbols: vscode.DocumentSymbol[] = ( // SymbolInformation has less detail (no children)
+            (await vscode.commands.executeCommand<vscode.DocumentSymbol[]>(
+                'vscode.executeDocumentSymbolProvider',
+                uri
+            )) || []
+        )
+
+        return symbols
+            .filter(sym => sym.kind === vscode.SymbolKind.Function)
+            .map(symbol => {
+                log(`found symbol: ${path.parse(filename).name}.${symbol.name}`)
+
+                return {
+                    filename,
+                    handlerName: `${path.parse(filename).name}.${symbol.name}`,
+                    positionStart: symbol.range.start.line,
```

Comment:
I agree and am not a fan of the confusing `positionStart` and `positionEnd`. I favor removing them in favor of `range` which is typically used by vscode. I left them as is to avoid going too crazy with re-writing `search.findCandidateLambdaHandlers`.

Added TODO.
✔️